### PR TITLE
BufferOverflow ServerCrash fix

### PR DESCRIPTION
**Pull request by timzijl**
Added extra checks for input to prefend servercrash because of bufferoverflow.

### CODE REVIEW

File: src/nl/hanze/gameserver/server/ClientInputHandler.java [hunk 1]
```diff
@@ -64,7 +66,14 @@ public void run() {
 				
 				// Get client input buffer and add data
 				ByteBuffer inputBuffer = getInputBuffer(client);
-				inputBuffer.put(data);
+                try {
+                    inputBuffer.put(data);
+                }
+                catch (BufferOverflowException bf) {
+                    client.writeResponse(new ErrorResponse("Username too long"));
```
Comment (goto-bus-stop) [inline src/nl/hanze/gameserver/server/ClientInputHandler.java]: This checks _every_ incoming message (which is great!), so it isn't necessarily the username causing issues. Perhaps the client is trying to `get gamelist AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA…`, for example.

A better error message could be something like "Message too large" or "Payload too large".

File: src/nl/hanze/gameserver/server/ClientInputHandler.java [hunk 2]
```diff
@@ -64,7 +66,14 @@ public void run() {
...
+                    client.disconnect();
+                    continue;
+                }
```
Comment (goto-bus-stop) [inline src/nl/hanze/gameserver/server/ClientInputHandler.java]: Indentation should be Tabs

**Comment by goto-bus-stop** [comment]
:tada: LGTM minus two small nits!

**Comment by leonwetzel** [comment]
Changes are committed.

**Comment by goto-bus-stop** [comment]
:100: thanks!

**Pull request merged**